### ISSUE TITLE
[Lab messages] Using the 'Person search' option when creating new entities, via processing lab messages, creates a new person 

### ISSUE THREAD

**Issue by adinaflorea9** [hzi-braunschweig/SORMAS-Project#8013]
### Bug Description

When processing lab messages to create new entities (cases and contacts), if the user uses the 'Person Search' option, the newly created entity is not associated with the selected person, but instead a new person gets created.

### Steps to Reproduce

1. Create a person by creating a new entity (i.e. 'person1');
2. Navigate to 'Lab messages' in the 'Samples' directory and click 'Process' on an unprocessed lab message;
3. In the first pop-up for 'Pick or create person' select 'Create new person' and click 'Confirm';
4. In the second pop-up for 'Pick or create entry' select 'Create new case' or 'Create new contact';
5. In the 'Create new case/contact' form, for the person related fields, use the 'Person search' option to select the person created at step1 ('person1');
6. Save the new entity (case or contact);
7. Navigate to the 'Persons' directory and observe the entities.

### Actual Behavior

There are 2 persons with the details of 'person1', but one of them is associated with the entity created at step1 and one of them is associated with the entity created at step6.

### Expected Behavior

There is 1 person with the details of 'person1' and it has 2 entities associated: the entity created at step1 and the entity created at step6.

### Screenshots

![image](https://user-images.githubusercontent.com/96909231/154064878-3a09f84d-3d3d-48de-afd5-c2028d9bbefb.png)

![image](https://user-images.githubusercontent.com/96909231/154064901-9e53079c-981b-4c74-b810-928a47e4d641.png)

### System Details

* Device: Windows 10
* SORMAS version: 1.69.0-SNAPSHOT
* Android version/Browser: Chrome
* Server URL: test-de2.sormas.netzlink.com
* User Role: Admin, NatUser

### Additional Information

Note 1: When processing a lab message and choosing to create a new event participant, the behavior described in the ticket does not reproduce.

Note 2: This issue might be related to ticket #8012

**Comment by FredrikSchaeferVitagroup** [comment]
I am not sure whether the expected behaviour described above is the only reasonable option here.
When being in the entry (case/contact/event participant) creation form during processing a lab message, the user has already gone through a person selection. 
An alternative expected behaviour could be that the person search button does not show in those forms when processing a lab message. 
![image](https://user-images.githubusercontent.com/67001822/154413541-8ae316f3-4b5b-416f-a1e0-da223b718d8d.png)

**Comment by kwa20** [comment]
@adinaflorea9 @FredrikSchaeferVitagroup I think the problem here is that firstly, the person that was selected in the person search pop-up is not actually linked and secondly, the existing case is for the selected person is not detected as a duplicate.

The person search in lab message processing is actually quite crucial because often enough the duplicate detection is not reliable enough to detect actual duplicates and users are then forced to create them to get the lab message data elsewhere in the system. If not at least one person is suggested as a duplicate, they're also not able to search for a different person without this option.

When a person is selected in the search person pop-up, there is no duplicate detection trigger needed because a person has been selected that already exists. However, it needs to be linked instead of created again. For the subsequent entity, e.g. case, this person might also already have a duplicate which was not previously suggested because the person wasn't detected. At this point this should get triggered again for a selected person.

**Comment by FredrikSchaeferVitagroup** [comment]
@kwa20 So this is the workflow you describe:

1. Start processing a lab message
2. Create a person (because the correct existing person is not proposed)
3. Select one of the entry creation forms (e.g. case creation form)
4. Select another person there
5. (continue)

I don't like it, for two reasons.
First, the newly created person is saved after step 2. When 4. is done, it needs to be removed, which is cumbersome.
Second, it's quite an effort for the user to get the correct person selected.

How about
- [ ] hide the search person button in the entry creation forms when processing a lab message
- [ ] show it in the Select or Create Person dialog when processing a lab message instead

**Comment by kwa20** [comment]
@FredrikSchaeferVitagroup I think I like that general idea and it would support that users usually work from a person oriented perspective.

> show it in the Select or Create Person dialog when processing a lab message instead

This pop-up is only triggered when a duplicate is suggested and it already gives you the option to search for a different person. The main problem here is that this pop-up is not shown all the time, even if duplicates might exist the detection might not find and suggest it.

Maybe we need to initialize that pop-up all the time and force users to chose if they want to search for an existing person, select a found duplicate or create a new one.

**Comment by FredrikSchaeferVitagroup** [comment]
@kwa20 Good point. So how about this?
- [ ] hide the search person button in the entry (case/contact/EP) creation forms when processing a lab message.
- [ ] show it in the Pick or create person dialog when processing a lab message instead (as 1 below).
- [ ] always show the Pick or create Person dialog, no matter whether a possible duplicate was automatically detected or not.
- [ ] change the behaviour of the list of proposed person (2 below). Fill it with possible matches automatically (like it is now). When the list is empty, hide it. Instead show a label stating: `No possible match could automatically be determined. You can manually search for existing persons.` Whatever person gets select by the manual search, add it to the list of proposed persons and select it (so that users can directly click on the `CONFIRM` button).
- [ ] change the caption of the option from `Select a matching person` to `Select an existing person`.

![image](https://user-images.githubusercontent.com/67001822/154438308-36ea564d-e4d9-4db6-9d42-c33aefaa7e15.png)

**Comment by kwa20** [comment]
@FredrikSchaeferVitagroup Sounds like a good plan!

Just some minor comments: 
> change the behaviour of the list of proposed person (2 below). Fill it with possible matches automatically (like it is now). When the list is empty, hide it. Instead show a label stating: No possible match could automatically be determined. You can manually search for existing persons. Whatever person gets select by the manual search, add it to the list of proposed persons and select it (so that users can directly click on the CONFIRM button).

Might be nice to have it as a little info icon plus text

> change the caption of the option from Select a matching person to Select an existing person.

The current caption is fine I think and would then be consistent with the other pop-ups

**Comment by FredrikSchaeferVitagroup** [comment]
Looking at the list of comments trying to specify this, I am pretty certain that this is not a bug, but a change request. I will change the ticket labels accordingly. This also means that it will not necessarily be done in this sprint.

**Comment by FredrikSchaeferVitagroup** [comment]
Quickly resolving the problem described in the summary (e.g. by hiding the search person button) could be a bugfix.

**Comment by kwa20** [comment]
@FredrikSchaeferVitagroup @adinaflorea9 I think what we need to fix here at least is the correct linking of the selected person. Otherwise, the implementation of https://github.com/hzi-braunschweig/SORMAS-Project/issues/6599 will be discarded for lab messages and that would be very unfortunate.

I think we can agree that the process is not ideal as descried in https://github.com/hzi-braunschweig/SORMAS-Project/issues/8013#issuecomment-1042669359. Therefore, I suggest to fix the person link with the bugfix so the implemented feature can be used and create a change request to implement what is described here https://github.com/hzi-braunschweig/SORMAS-Project/issues/8013#issuecomment-1042711748

**Comment by FredrikSchaeferVitagroup** [comment]
Really fixing this is not so easy. [As I wrote before](https://github.com/hzi-braunschweig/SORMAS-Project/issues/8013#issuecomment-1042669359), the duplicate person gets created and saved before the user selects another one.

**Comment by FredrikSchaeferVitagroup** [comment]
As @kwa20 pointed out, it is necessary that the pick or create person dialog always shows when processing lab messages then, not only when there are possible matches. This should also be be implemented in this ticket.

**Comment by abrudanancuta** [comment]
Validated on local environment with version 1.69.0-snapshot
